### PR TITLE
chore: move to pyproject.toml, add pre-commit, move to ruff

### PR DESCRIPTION
**Pull request by wolfv**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+dynamic = ["dependencies"]
+
+[tool.setuptools]
+packages = ["webservices_dispatch_action"]
+
+[project.scripts]
+run-webservices-dispatch-action = "webservices_dispatch_action.__main__:main"
+run-webservices-dispatch-action-version-updater = "webservices_dispatch_action.version_updater:main"
+
+[tool.flake8]
```
Comment (beckermr) [inline pyproject.toml]: This won't work IIRC. See https://pypi.org/project/Flake8-pyproject/

**Review by beckermr — CHANGES_REQUESTED** [review]
One issue with the flake8 setup.

File: pyproject.toml (hunk 1 above)
Comment (wolfv) [reply pyproject.toml]: OK to move to ruff?

Comment (beckermr) [reply pyproject.toml]: Yes OFC! We should add pre-commit as well.

File: pyproject.toml [hunk 2]
```diff
@@ -14,5 +14,5 @@ packages = ["webservices_dispatch_action"]
 run-webservices-dispatch-action = "webservices_dispatch_action.__main__:main"
 run-webservices-dispatch-action-version-updater = "webservices_dispatch_action.version_updater:main"
 
-[tool.flake8]
-max-line-length = 88
\ No newline at end of file
+[tool.ruff]
+line-length = 88
```
Comment (beckermr) [inline pyproject.toml]:
```suggestion
[tool.ruff.lint]
select = ["E", "F", "I", "W"]
preview = true

[tool.ruff.lint.pycodestyle]
max-line-length = 88
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -33,13 +33,13 @@ jobs:
...
           conda info -a
-          mamba install --yes pytest flake8
+          mamba install --yes pytest pre-commit
           pip install --no-deps -e .
 
       - name: lint and run tests
         shell: bash -l {0}
         run: |
-          flake8 webservices_dispatch_action
+          pre-commit --run all
```
Comment (beckermr) [inline .github/workflows/tests.yml]:
```suggestion
          pre-commit run --all
```

Comment (wolfv) [reply .github/workflows/tests.yml]: Good catch. I ran this with my pixi.toml haha :)

**Comment by beckermr** [comment]
pre-commit.ci autofix

**Comment by beckermr** [comment]
I enabled pre-commit.ci for this repo. IDK how long it takes to catchup.

**Comment by beckermr** [comment]
🎉 We can remove the linting from the tests too.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -33,13 +33,13 @@ jobs:
         shell: bash -l {0}
         run: |
           conda info -a
-          mamba install --yes pytest flake8
+          mamba install --yes pytest pre-commit
```
Comment (beckermr) [inline .github/workflows/tests.yml]:
```suggestion
          mamba install --yes pytest
```

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -33,13 +33,13 @@ jobs:
...
           pip install --no-deps -e .
 
       - name: lint and run tests
         shell: bash -l {0}
         run: |
-          flake8 webservices_dispatch_action
+          pre-commit run --all
```
Comment (beckermr) [inline .github/workflows/tests.yml]:
```suggestion
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -33,13 +33,12 @@ jobs:
         shell: bash -l {0}
         run: |
           conda info -a
-          mamba install --yes pytest flake8
+          mamba install --yes pytest
           pip install --no-deps -e .
 
       - name: lint and run tests
```
Comment (beckermr) [inline .github/workflows/tests.yml]:
```suggestion
      - name: run tests
```

**Comment by wolfv** [comment]
Cool. With pixi, we'd also have an easy way to run it locally.

**Comment by beckermr** [comment]
Also `pre-commit run -a` is your friend.

**Comment by wolfv** [comment]
feel free to merge

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
cool! thanks @wolfv for the updates!

**Pull request merged**